packages/core/core-flows/src/order/workflows/transfer/decline-order-transfer.ts

```diff
@@ -0,0 +1,87 @@
+import {
+  OrderChangeDTO,
+  OrderDTO,
+  OrderWorkflow,
+} from "@medusajs/framework/types"
+import {
+  ChangeActionType,
+  MedusaError,
+  OrderChangeStatus,
+} from "@medusajs/framework/utils"
+import {
+  WorkflowData,
+  createStep,
+  createWorkflow,
+} from "@medusajs/framework/workflows-sdk"
+
+import { useRemoteQueryStep } from "../../../common"
+import { declineOrderChangeStep } from "../../steps"
+import {
+  throwIfIsCancelled,
+  throwIfOrderChangeIsNotActive,
+} from "../../utils/order-validation"
+
+/**
+ * This step validates that a requested order transfer can be declineed.
+ */
+export const declineTransferOrderRequestValidationStep = createStep(
+  "validate-decline-transfer-order-request",
+  async function ({
+    order,
+    orderChange,
+    input,
+  }: {
+    order: OrderDTO
+    orderChange: OrderChangeDTO
+    input: OrderWorkflow.DeclineTransferOrderRequestWorkflowInput
+  }) {
+    throwIfIsCancelled(order, "Order")
+    throwIfOrderChangeIsNotActive({ orderChange })
+
+    const token = orderChange.actions?.find(
+      (a) => a.action === ChangeActionType.TRANSFER_CUSTOMER
+    )?.details!.token
+
+    if (!input.token.length || token !== input.token) {
```

Comment:
to account for undefined cases:

```suggestion
    if (!input.token?.length || token !== input.token) {
```